azure_costs_exporter/prometheus_collector.py

```diff
@@ -65,9 +66,15 @@ def _get_azure_data(self, month=None):
         url = "https://ea.azure.com/rest/{0}/usage-report?month={1}&type=detail&fmt=Json".format(self._enrollment,
                                                                                                  month)
 
-        rsp = requests.get(url, headers=headers, timeout=10)
-        rsp.raise_for_status()
+        try:
+            timeout = int(float(request.headers.get('Scrape-Timeout-Seconds', 10)))
```

Comment:
Ah, I saw `Scrape-Timeout-Seconds` in one of the prometheus issues, thanks.

Regarding conversion: I don't know, it was quite late already :)